### PR TITLE
[WNMGDS-397] Create testing scripts

### PR DESCRIPTION
**Pull request by bernardwang**
## Summary
- Create `yarn test` and `yarn test:e2e` cli commands
- Update all test files and snapshots

## How to test
- Run `yarn test` to test unit tests.
- Run `yarn test:e2e` to test e2e tests, a few will fail but I think we can handle these in a follow up PR
- Test various cli options like  "--skipBuild", "--updateSnapshots" 
- Test on a child DS.
[child-design-system-example.zip](https://github.com/CMSgov/design-system/files/4589444/child-design-system-example.zip)

### CODE REVIEW

File: packages/design-system-scripts/jest/unit.config.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
+const path = require('path');
+
+module.exports = (rootDir) => ({
+  rootDir,
+  testURL: 'http://localhost',
+  setupFiles: [`<rootDir>/${path.relative(rootDir, __dirname)}/polyfills.js`],
+  setupFilesAfterEnv: [`<rootDir>/${path.relative(rootDir, __dirname)}/setupEnzyme.js`],
```
Comment (bernardwang) [inline packages/design-system-scripts/jest/unit.config.js:7]: I keep getting the following Enzyme error:
```
Enzyme Internal Error: Enzyme expects an adapter to be configured, but found none.
          To configure an adapter, you should call `Enzyme.configure({ adapter: new Adapter() })`
          before using any of Enzyme's top level APIs, where `Adapter` is the adapter
          corresponding to the library currently being tested. For example:

          import Adapter from 'enzyme-adapter-react-15';

          To find out more about this, see http://airbnb.io/enzyme/docs/installation/index.html
```
When I copy the enzyme/adapter config from `setupEnzyme.js` to the top of a test file, the error goes away. Frustratingly, if I add a console.log to `setupEnzyme.js` I can see that the file is being found and run before each test file like its supposed to, but some reason I'm still getting the Enzyme error 😠

Comment (bernardwang) [reply packages/design-system-scripts/jest/unit.config.js]: I couldnt completely solve the issue, the solution im going with for now is to move the `setupEnzyme` file to the root of the directory

File: packages/design-system-docs/src/scripts/helpers/githubUrl.js [hunk 2]
```diff
@@ -1,7 +1,10 @@
-import path from 'path';
+import join from 'url-join';
```
Comment (bernardwang) [inline packages/design-system-docs/src/scripts/helpers/githubUrl.js:1]: Added `url-join` as a dep to help with this helper func

File: packages/design-system-scripts/cli.js [hunk 3]
```diff
@@ -85,6 +84,70 @@ yargs
...
+        '--config',
+        JSON.stringify(unitConfig(argv.directory)),
+        ...(argv.updateSnapshot ? ['--updateSnapshot'] : []),
+        ...(argv.watch ? ['--watch'] : []),
+        ...(argv.ci ? ['--ci'] : []),
+      ]);
+    },
+  })
+  .command({
+    command: 'test:e2e <directory>',
```
Comment (bernardwang) [inline packages/design-system-scripts/cli.js:119]: Separate command for e2e because it requires unique options

**Comment by bernardwang** [comment]
Tests are failing in Travis CI, but should be working locally. I'll look into fixing travis now but I don't think that should block reviews

**Comment by line47** [comment]
I'm wondering if we should update the scripts section of the README with CLI options for
`--updateSnapshots`, `--skipBuild`, and any others that may be new  https://github.com/CMSgov/design-system#scripts

**Comment by bernardwang** [comment]
@line47 I was thinking of doing all those documentation updates all at once here: https://github.com/CMSgov/design-system/pull/666
as currently none of our documentation has been updated with the new scripts

**Comment by bernardwang** [comment]
I saw that `yarn start` is resulting in this error: `Error: Cannot find module '@cmsgov/design-system'`. Working on fixing it now

**Comment by nichia** [comment]
Core Design System:
- `yarn install` ✅
- `yarn build` ✅
- `yarn test` ✅completed with 2 failed test suites , 1 failed tests *
- `yarn test:e2e` ✅completed with 2 failed test suites , 3 failed tests *
- `yarn --updateSnapshots` ✅
- `yarn --skipBuild` ✅
* failed test suites/tests does not affect this PR - these failed tests will be looked at in separate PR

Child Design System:
- `yarn install` ✅
- `yarn build` ❌
- `yarn test` ✅completed with 2 failed test suites , 11 failed tests
- `yarn test:e2e`  ❌ error `Command "test:e2e" not found.` 
- `yarn --updateSnapshots` ✅
- `yarn --skipBuild` ✅

**Comment by bernardwang** [comment]
@nichia For the child DS `yarn test:e2e` wont work because I didnt add that to the package.json. I didn't bother because we don't have any e2e tests for the child ds. You can run it anyway with `yarn cmsds test:e2e` but it wont do anything.

**Comment by line47** [comment]
**Core Design System:**

- yarn install ✅
- yarn build ✅
- yarn test ✅
- yarn test:e2e ✅ completed with 2 failed test suites , 3 failed tests *
- yarn --updateSnapshots ✅
- yarn --skipBuild ✅

**Child Design System:**

- yarn install ✅
- yarn build 🤔 completed but got a lot of messages about ` Field 'browser' doesn't contain a valid alias configuration as directory` 
- yarn test ✅completed with 2 failed test suites
- yarn test:e2e ❌ error Command "test:e2e" not found.
- yarn --updateSnapshots ✅
- yarn --skipBuild ✅

**Pull request merged**